src/categoricalvector.jl

```diff
@@ -43,17 +43,13 @@ struct CategoricalVector{T, A<:AbstractVector{T}} <: AbstractVector{T}
     data::A
 end
 
-function CategoricalVector(data::AbstractVector{T}) where T
-    CategoricalVector{T, typeof(data)}(data)
-end
```

Comment:
the default constructor takes care of this. so this caused a redefinition warning.